core/src/main/scala/org/apache/spark/sql/delta/PreprocessTableWithDVs.scala

```diff
@@ -84,9 +84,6 @@ object ScanWithDeletionVectors {
       hadoopRelation: HadoopFsRelation,
       fileFormat: DeltaParquetFileFormat,
       index: TahoeFileIndex): Option[LogicalPlan] = {
-    // If the table has no DVs enabled, no change needed
-    if (!deletionVectorsReadable(index.protocol, index.metadata)) return None
-
     require(!index.isInstanceOf[TahoeLogFileIndex],
       "Cannot work with a non-pinned table snapshot of the TahoeFileIndex")
 
```

Comment:
while this is indeed redundant...there is difference between keeping the copy before or after the require. I am not which one is the right thing to do. @larsk-db @vkorukanti can you provide some clarity?